mesonbuild/build.py

```diff
@@ -451,15 +451,22 @@ def get_outputs(self, backend: 'Backend') -> T.List[str]:
             for source in self.get_sources(self.srclist, self.genlist)
         ]
 
+EnvInitValueType = T.Dict[str, T.Union[str, T.List[str]]]
+
 class EnvironmentVariables(HoldableObject):
-    def __init__(self, values: T.Optional[T.Dict[str, str]] = None) -> None:
+    def __init__(self, values: T.Optional[EnvValueType] = None,
+                 init_method: str = 'set', separator: str = os.pathsep) -> None:
```

Comment:
```suggestion
                 init_method: Literal['set', 'prepend', 'append'] = 'set', separator: str = os.pathsep) -> None:
```

Which either needs to be string declared, or in a `__future__.annotations` module, since it's still from `typing_extensions`. This will get mypy to catch any values that are not in the expected values.

Reply:
done.